tests/test_user_util.py

```diff
@@ -217,4 +217,66 @@ def test_username_to_hash_bad_salt(salt):
     Salts that are *not* lists/tuples should fail.
     """
     with pytest.raises((ValueError, IndexError)):
-        retired_username = user_util.get_retired_username('AnotherLearnerUserName', salt)
+        _ = user_util.get_retired_username('AnotherLearnerUserName', salt)
+
+
+#
+# External user retirement tests
+#
+
+@pytest.mark.parametrize('salt_list', VALID_SALT_LISTS)
+def test_external_key_to_hash(salt_list):
+    external_key = '343ni3hr3ifh3fgghg'
+    retired_external_key = user_util.get_retired_external_key(external_key, salt_list)
+    assert retired_external_key != external_key
+    assert retired_external_key.startswith(
+        '_'.join(user_util.RETIRED_EXTERNAL_KEY_DEFAULT_FMT.split('_')[0:3])
+    )
+    # Since SHA1 is used, the hexadecimal digest length should be 40.
+    assert len(retired_external_key) == len('retired_external_key_') + 40
+
+
+def test_unicode_external_key_to_hash():
+    unicode_external_key = u'🅐.🅛🅔🅐🅡🅝🅔🅡'
+    retired_external_key= user_util.get_retired_external_key(unicode_external_key, VALID_SALT_LIST_ONE_SALT)
+    assert retired_external_key != unicode_external_key
+    # Since SHA1 is used, the hexadecimal digest length should be 40.
+    assert len(retired_external_key) == len('retired_external_key_') + 40
+
+
+@pytest.mark.parametrize('salt_list', (VALID_SALT_LIST_THREE_SALTS,))
+def test_correct_external_key_hash(salt_list):
+    """
+    Verify that get_retired_external_key uses the current salt and returns the expected hash.
+    """
+    external_key = 'S34839GEF3'
+    valid_retired_external_keys = [
+        user_util.RETIRED_EXTERNAL_KEY_DEFAULT_FMT.format(
+            user_util._compute_retired_hash(external_key.lower(), salt)
+        )
+        for salt in salt_list
+    ]
+    retired_email = user_util.get_retired_external_key(external_key, salt_list)
+    assert retired_email == valid_retired_external_keys[-1]
+
+
+@pytest.mark.parametrize('salt_list', (VALID_SALT_LIST_FIVE_SALTS,))
+def test_all_external_keys_to_hash(salt_list):
+    external_key = 'S34839GEF3'
+    retired_external_key_generator = user_util.get_all_retired_external_keys(external_key, salt_list)
+    assert isinstance(retired_external_key_generator, GeneratorType)
+    assert len(list(retired_external_key_generator)) == len(VALID_SALT_LIST_FIVE_SALTS)
+
+
+@pytest.mark.parametrize('salt_list', VALID_SALT_LISTS)
+def test_external_key_to_hash_with_different_format(salt_list):
+    external_key = 'S34839GEF3'
+    retired_external_key_fmt = "{}_is_now_the_retired_external_key"
+    retired_external_key = user_util.get_retired_external_key(
+        external_key,
+        salt_list,
+        retired_external_key_fmt=retired_external_key_fmt
+    )
+    assert retired_external_key.endswith('_'.join(retired_external_key_fmt.split('_')[1:]))
```

Comment:
Not a blocker, and I know it's present elsewhere in the repo, but I think it would be clearer to have the assertions check against simple values. It is easier to read and debug something like:
```
assert retired_external_key.endswith('_is_now_the_retired_external_key')
```

Reply:
Thank you for the suggestion. I updated the test case.